src/App/Fossa/EmbeddedBinary.hs

```diff
@@ -134,10 +156,21 @@ extractedPath bin = case bin of
   Themis -> $(mkRelFile "themis-cli")
   ThemisIndex -> $(mkRelFile "index.gob.xz")
 
+-- | Extract to @$TMP/fossa-vendor/<timestamp>
+-- We used to extract everything to @$TMP/fossa-vendor@, but there's a subtle issue with that.
+-- Cleanup is just removing the directory where the file resides, which is fine unless there's
+-- more than one active extracted file.  Cleanup could potentially kill both while one is in use.
+-- Extracting to a another subdir meane that the cleanup only cleans the timestamp subdir.
+-- The only downside is that we never cleanup the fossa-vendor directory, which is not an issue,
+-- since it should be empty by the time we finish cleanup.  The tempfile cleaner on the system
+-- should pick it up anyway.
 extractDir :: Has (Lift IO) sig m => m (Path Abs Dir)
 extractDir = do
   wd <- sendIO getTempDir
-  pure (wd </> $(mkRelDir "fossa-vendor"))
+  -- Get some positive "random" number, in this case a timestamp
+  ts <- show @Int . abs . floor <$> sendIO getPOSIXTime
```

Comment:
Do need to worry about the two dirs having the same timestamp? `getPOSIXTime` has a resolution of microseconds, but then applying floor to it, so I think we only have a resolution of seconds.

The easy fix is to multiply it by 1,000,000 before `floor`

Reply:
Nice catch.